### DISCUSSION TITLE
Using .env file with PDM

### DISCUSSION THREAD

**Discussion by ksze** [pdm-project/pdm#660] [Q&A] [answered]
In old school virtual envs, one could add more env vars in to the activate script.
With pipenv, `pipenv shell` and `pipenv run` would also automatically load a .env file.

PDM doesn't seem to load a .env file when I do, for exacmple, `pdm run env`.

In the PDM world, is there a canonical way to configure the environment using a .env file or something similar?

**Comment by ksze** [comment, marked as answer]
Never mind. I just found this: https://pdm.fming.dev/usage/project/#environment-variables-support

**Comment by cye18** [comment]
I had the same question and came across this discussion. The link @ksze provided seems to be invalid in later releases. I found the link to release 1.8.5 that still has the content.

https://github.com/pdm-project/pdm/blob/1.8.5/docs/docs/usage/project.md#environment-variables-support

**Reply by frostming** [reply]
Documentation is updated since then, the new link: https://pdm.fming.dev/usage/scripts/#env

Or you can take advantage of the search box in the navbar.

**Reply by cye18** [reply]
Oh yes, I didn't realize there is a search box. Thanks!

**Comment by Fak3** [comment]
related issue: https://github.com/abersheeran/pdm-shell/issues/1

**Comment by vipinmohan22** [comment]
for all who are coming here after august 2024.. you can set default env for all pdm run commands using the following snippet

```
[tool.pdm.scripts]
...
_.env_file = ".env"
start.cmd = "..."
....
```
